release/run_e2e.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+set -x
+
+cd "${0%/*}" || exit 1
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+case $key in
+    --ray-repo)
+    shift
+    RAY_REPO=$1
+    ;;
+    --ray-branch)
+    shift
+    RAY_BRANCH=$1
+    ;;
+    --ray-version)
+    shift
+    RAY_VERSION=$1
+    ;;
+    --ray-wheels)
+    shift
+    RAY_WHEELS=$1
+    ;;
+    --ray-test-repo)
+    shift
+    RAY_TEST_REPO=$1
+    ;;
+    --ray-test-branch)
+    shift
+    RAY_TEST_BRANCH=$1
+    ;;
+    --release-results-dir)
+    shift
+    RELEASE_RESULTS_DIR=$1
+    ;;
+    *)
+    break
+esac
+shift
+done
+
+RAY_TEST_REPO=${RAY_TEST_REPO-https://github.com/ray-project/ray.git}
+RAY_TEST_BRANCH=${RAY_TEST_BRANCH-master}
+
+export RAY_REPO RAY_BRANCH RAY_VERSION RAY_WHEELS RAY_TEST_REPO RAY_TEST_BRANCH RELEASE_RESULTS_DIR
+
+pip install -q -r requirements.txt
+pip install -U boto3 botocore
+git clone -b "${RAY_TEST_BRANCH}" "${RAY_TEST_REPO}" ~/ray
+
+RETRY_NUM=1
+MAX_RETRIES=${MAX_RETRIES-3}
+
+if [ "${BUILDKITE_RETRY_COUNT-0}" -ge 1 ]; then
+  echo "This is a manually triggered retry from the Buildkite web UI, so we set the number of infra retries to 1."
+  MAX_RETRIES=1
+fi
+
+while [ "$RETRY_NUM" -le "$MAX_RETRIES" ]; do
+  if [ "$RETRY_NUM" -gt 1 ]; then
+    SLEEP_TIME=$((600 * (RETRY_NUM - 1)))
```

Comment:
Maybe adding a logic to fail the test if it runs more than X hours? I think if it lasts longer than that, it can actually influence other tests in a bad way.

Reply:
We can add a buildkite option for that, but we enforce timeouts pretty rigorously already, so we don't necessarily need that.
Restarts only happen for infra errors, so before any kind of long running test is started